src/MyEventEmitter.js

```diff
@@ -1,14 +1,70 @@
 'use strict';
 
 class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  constructor() {
+    this.listeners = [];
+  }
+
+  on(eventName, listener) {
+    const listenerObj = { name: eventName, function: listener, once: false };
+
+    this.listeners.push(listenerObj);
+  }
+
+  once(eventName, listener) {
+    const listenerObj = { name: eventName, function: listener, once: true };
+
+    this.listeners.push(listenerObj);
+  }
+
+  off(eventName, listener) {
+    this.listeners = [...this.listeners].filter(
+      (obj) => !(obj.name === eventName && obj.function === listener),
+    );
```

Comment:
The use of the spread operator here creates a shallow copy of the listeners array, which is not necessary for filtering. You can directly filter the `this.listeners` array without creating a new array, which will be more efficient.